### ISSUE TITLE
Very high CPU usage when repeatedly restarting containers and monitoring their stats

### ISSUE THREAD

**Issue by AndreyKostov** [docker/docker#17636]
Description of problem:

When rapidly restarting containers (two containers once every minute to be precise) and beginning monitoring them at the same time with docker stats each time we spawn a new one, over time the engine consumes more and more cpu (and memory but that one is not as problematic)

`docker version`:

1.9.0rc4

`docker info`:

```
Containers: 13
Images: 92
Server Version: 1.9.0-cs1-rc1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 118
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-44-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 1
Total Memory: 3.676 GiB
Name: dhetemp-u1404-20151102-214124-8579-rm-after-2015-11-04
ID: UIRZ:XBXD:5ISE:OHZD:MTU6:QPUV:XSNV:5BPY:3ABQ:UB4X:LRWQ:BPVM
Username: dockerautomation
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Labels:
 provider=amazonec2
```

`uname -a`:

```Linux dhetemp-u1404-20151102-214124-8579-rm-after-2015-11-04 3.13.0-44-generic #73-Ubuntu SMP Tue Dec 16 00:22:43 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux```

Environment details (AWS, VirtualBox, physical, etc.):
AWS

How reproducible:
Call docker stats many times on a container and eventually you are left a lot of goroutines doing stats calls (as evidenced by the stack trace I have attached)

Actual Results:
50-60% cpu usage (growing into it slowly over time)

Expected Results:
NOT 50-60% cpu usage with no new api calls

Additional info:

Here's a stack trace generated by doing a ```kill -s USR1```

https://gist.github.com/AndreyKostov/a84e8eb6c0836be0e74f

**Comment by crosbymichael** [comment]
This seems to happen with multiple docker stats calls

**Comment by crosbymichael** [comment]
Its defiantly stats released in 1.9.

I tested 1.7, 1.8, and 1.9.

Results for 2 containers and 5 docker stats calls:

1.7 - 1.9% cpu
1.8 ( https://github.com/docker/docker/commit/8b40e44c395fb307cceb6f179ffe02030b316535 ) - 10-30% cpu
1.9 - 30-60% cpu

**Comment by calavera** [comment]
#17638 should fix this issue.
